app/src/components/avatar_event_bus.rs

```diff
@@ -0,0 +1,55 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use yew_agent::{Agent, AgentLink, Context, HandlerId};
+
+use super::avatar::AvatarData;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Request {
+    Update((String, Option<AvatarData>)),
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Response {
+    Update((String, Option<AvatarData>)),
+}
+
+pub struct AvatarEventBus {
+    link: AgentLink<AvatarEventBus>,
+    subscribers: HashSet<HandlerId>,
+}
+
+impl Agent for AvatarEventBus {
```

Comment:
Maybe we can keep a map of subscribers per username? So we only send the update to the right component, based on what they requested.

Reply:
It would have to be a double map:
UserId to Vec<HandlerId>, and HandlerId to Vec<UserId> for cleanup.
Maybe it's overkill? WDYT ?